### PR TITLE
feat: Send fee to fee collector

### LINKED ISSUE #28: When a name is registered, send the fee to the fee collector
@fzavalia commented on [Mon Mar 06 2023](https://github.com/decentraland/account/issues/257)

Instead of burning it.

Include tests.

### PR DESCRIPTION
**Pull request by fzavalia**
Closes #28 

Instead of transferring the fee from the caller to the contract and then burning it, transfer the fee to the fee collector directly.

### CODE REVIEW

File: test/SubdomainWithDCLControllerV2.spec.js [hunk 1]
```diff
@@ -2001,7 +2001,7 @@ describe('DCL Names V2 with DCLControllerV2', function () {
           { ...fromUser, gasPrice: MAX_GAS_PRICE }
         )
 
-        expect(logs.length).to.be.equal(5)
+        expect(logs.length).to.be.equal(4)
```
Comment (fzavalia) [inline test/SubdomainWithDCLControllerV2.spec.js:2004]: There is no more Burn event being emmited

File: test/SubdomainWithDCLControllerV2.spec.js [hunk 2]
```diff
@@ -2059,6 +2054,24 @@ describe('DCL Names V2 with DCLControllerV2', function () {
         expect(currentResolver).to.be.equal(ZERO_ADDRESS)
       })
 
+      it('should transfer the fee from the caller to the fee collector', async function () {
+        const userBalance = await manaContract.balanceOf(user)
+        const feeCollectorBalance = await manaContract.balanceOf(feeCollector)
```
Comment (nachomazzara) [inline test/SubdomainWithDCLControllerV2.spec.js:2059]: expect current balance?
and expect user balance?

File: contracts/ens/DCLControllerV2.sol [hunk 3]
```diff
@@ -69,10 +69,8 @@ contract DCLControllerV2 is Ownable {
 
         // Register the name
         registrar.register(_name, _beneficiary);
-        // Debit `price` from sender
-        acceptedToken.transferFrom(msg.sender, address(this), PRICE);
-        // Burn it
-        acceptedToken.burn(PRICE);
+        // Transfer the fee to the fee collector
```
Comment (nachomazzara) [inline contracts/ens/DCLControllerV2.sol:72]:
```suggestion
        // Transfer PRICE to the fee collector
```

File: test/SubdomainWithDCLControllerV2.spec.js (hunk 2 above)
Comment (fzavalia) [reply test/SubdomainWithDCLControllerV2.spec.js:2063]: added price validation

**Review by nachomazzara — APPROVED** [review]

**Pull request merged**